config/watcher.go

```diff
@@ -56,6 +57,16 @@ func newCloudWatcher(ctx context.Context, config *Config, logger logging.Logger)
 
 	ticker := time.NewTicker(config.Cloud.RefreshInterval)
 
+	// TODO: the watcher should also restablish this connection.
+	conn, err := CreateNewGRPCClient(ctx, config.Cloud, logger)
+	if err != nil {
+		cancel()
+		logger.Warn("failed to connect to app - cannot watch for config updates")
+		return nil
+	}
+	defer utils.UncheckedErrorFunc(conn.Close)
+	rr := remoteReader{apppb.NewRobotServiceClient(conn)}
+
```

Comment:
this is a significant behavior change - before we were creating connections on demand closing them within the loop - here we just create a connection once. the biggest question is how we handle a connection failure.

Reply:
can we nil out the client on every error and then remake the connection the next time around?